flink-end-to-end-tests/test-scripts/common_yarn_docker.sh

```diff
@@ -100,13 +100,10 @@ function build_image() {
     echo "Pre-downloading Hadoop tarball"
     local cache_path
     cache_path=$(get_artifact "http://archive.apache.org/dist/hadoop/common/hadoop-2.8.5/hadoop-2.8.5.tar.gz")
-    ln "${cache_path}" "${END_TO_END_DIR}/test-scripts/docker-hadoop-secure-cluster/hadoop-2.8.5.tar.gz"
+    ln "${cache_path}" "${END_TO_END_DIR}/test-scripts/docker-hadoop-secure-cluster/hadoop/hadoop.tar.gz"
```

Comment:
IIRC the danger of this is that now any Hadoop version will get cached, because there's no version number anymore. That means that if an upgrade is done from Hadoop 2.8.5 to 2.10.2, no new Hadoop TAR will be downloaded because it will be cached from a previous run.

Reply:
I thought the cache is per run. OK, then putting back the version here too...

Reply:
Wait a minute, `get_artifact` downloads the jar if needed and we just make a link to the cache entry.
[Kafka](https://github.com/apache/flink/blob/master/flink-end-to-end-tests/test-scripts/kafka-common.sh#L43) works the same way where several version bump already happened.

Reply:
> I thought the cache is per run. OK, then putting back the version here too...

That would limit the caching benefits significantly because many things are only required once.

> Wait a minute, get_artifact downloads the jar if needed and we just make a link to the cache entry.
[Kafka](https://github.com/apache/flink/blob/master/flink-end-to-end-tests/test-scripts/kafka-common.sh#L43) works the same way where several version bump already happened.

The code as-is in the PR is fine. The important thing here is the URL passed to `get_artifact` contains the version, since we only create a symbolic link to the path returned by `get_artifact` (which also contains the version, extracted from the URL).
If you were to _copy_ the file to `cache_path` then @MartijnVisser would be correct though.